src/com/etendoerp/dependencymanager/util/PackageUtil.java

```diff
@@ -221,6 +224,62 @@ public static synchronized void updateOrCreateDependency(String group, String ar
     OBDal.getInstance().save(existingDependency);
   }
 
+  /**
+   * Fetches the latest package version from the database using OBCriteria.
+   *
+   * @param group The group of the package.
+   * @param artifact The artifact of the package.
+   * @return The latest version string or null if no versions are found.
+   */
+  private static String getLastPackageVersionFromDB(String group, String artifact) {
+    OBCriteria<PackageVersion> criteria = OBDal.getInstance().createCriteria(PackageVersion.class);
+    criteria.createAlias("package", "pkg");
+    criteria.add(Restrictions.eq("pkg.artifact", artifact));
+    criteria.add(Restrictions.eq("pkg.group", group));
+    criteria.addOrder(Order.desc(VERSION));
+    criteria.setMaxResults(1).uniqueResult();
+
+    List<PackageVersion> packageVersions = criteria.list();
```

Comment:
???

Why are you setting a setMaxResults(1).uniqueResult() and after that you are using a list()??